pkg/cli/cli.go

```diff
@@ -983,23 +1000,166 @@ func printConnection(c *command, cvd RemoteCVDLocator, status ConnStatus) {
 	c.Printf("%s/%s: %s\n", cvd.Host, cvd.WebRTCDeviceID, state)
 }
 
-func buildAgentCmdArgs(flags *ConnectFlags, device string) []string {
-	args := []string{
-		ConnectionAgentCommandName,
-		device,
-	}
+func buildAgentCmdArgs(flags *ConnectFlags, device string, connectAgent string) []string {
+	args := []string{connectAgent, device}
 	return append(args, flags.AsArgs()...)
 }
 
-// Handler for the agent command. This is not meant to be called by the user
-// directly, but instead is started by the open command.
+// Return the ADB address of remote Cuttlefish instance as [host]:port.
+func findDeviceAdbAddress(service client.Service, controlDir, hostname, device string) (string, error) {
```

Comment:
Removed the logic introducing the new function`findDeviceAdbAddress`. OTOH, I thought introducing `adbPort` would't worth that much as well, since `adbPort(cvd *CVD)` will contain `net.SplitHostPort` only inside. I picked it up all logics into `runConnectionProxyAgentCommand` like before.